dotCMS/src/main/java/com/dotcms/variant/business/web/CurrentVariantWebInterceptor.java

```diff
@@ -0,0 +1,101 @@
+package com.dotcms.variant.business.web;
+
+import static com.dotcms.util.CollectionsUtils.list;
+
+import com.dotcms.filters.interceptor.Result;
+import com.dotcms.filters.interceptor.WebInterceptor;
+import com.dotcms.variant.VariantAPI;
+import com.dotmarketing.util.LoginMode;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.util.StringPool;
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+/**
+ * Listen and Intercept any request to:
+ *
+ * - /da/*
+ * - /contentAsset/*
+ *
+ * And check if the request has the parameter <code>variantName</code>, if the parameter not exists
+ * then check if the request has the header <code>referer</code> and if the header has the parameter
+ * <code>variantName</code> then set it as an attribute to the request.
+ */
+public class CurrentVariantWebInterceptor implements WebInterceptor {
+    private static final long serialVersionUID = 1L;
+    private static final String[] PATHS = new String[] {
+            "/dA/*",
+            "/contentAsset/*"
+    };
+
+    @Override
+    public String[] getFilters() {
+        return PATHS;
+    }
+
+    @Override
+    public Result intercept(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+        final String requestParameter = request.getParameter(VariantAPI.VARIANT_KEY);
+
+        if (UtilMethods.isSet(requestParameter)) {
+            return Result.NEXT;
+        }
+
+        String currentVariantName = null;
+        final String refererValue = request.getHeader("referer");
+
+        if (UtilMethods.isSet(refererValue)) {
+            final Optional<String> variantValueOptional = getVariantValueFromReferer(refererValue);
+
+            if (variantValueOptional.isPresent()) {
+                currentVariantName = variantValueOptional.get();
+            }
+        }
+
+        if (!UtilMethods.isSet(currentVariantName)) {
+            final HttpSession session = request.getSession();
+
+            if (session != null) {
+                final Object attribute = session.getAttribute(VariantAPI.VARIANT_KEY);
+
+                if (isValidSessionAtribute(attribute) && LoginMode.get() != LoginMode.FE){
+                    final CurrentVariantSessionItem currentVariantSessionItem = (CurrentVariantSessionItem) attribute;
+
+                    if (!currentVariantSessionItem.isExpired()) {
+                        currentVariantName = currentVariantSessionItem.getVariantName();
+                    }
+                }
+            }
+        }
+
+        if (UtilMethods.isSet(currentVariantName)) {
+            request.setAttribute(VariantAPI.VARIANT_KEY, currentVariantName);
+        }
+
+        return Result.NEXT;
+    }
+
+    private boolean isValidSessionAtribute(Object attribute) {
```

Comment:
Should this return a variant if there is no matching variant in the dotCMS system?  Meaning, should we go to the API to check the validity of the value here?

Reply:
do you mean that here 

https://github.com/dotCMS/core/pull/25387/files/711dec5264046ecef1df3dab93eb0e45e968ad72#diff-65fdb30fd885a08035709f661b6cd0e1e4f4354bde74a4f3e1f6464e765d0731R75

we should set the Variant object and check if it really exists?

Reply:
I am doing it here https://github.com/dotCMS/core/pull/25387/files/711dec5264046ecef1df3dab93eb0e45e968ad72#diff-c67db85ad5a8303546f73462e9074d4cf3f669295e99d9ca050c4bd8ce25e7f2R57